### PR TITLE
Create unit tests for cleanup_table

### PR DESCRIPTION
**Pull request by dimberman**
# Description
## What is the current behavior?
Currenlty the `test_cleanup` file tests take 16 minutes to run. This is due to many external database calls and file loads/table creations. This is causing us unnecessary expenses and slowing down development as we wait for CI/CD to finish.

## What is the new behavior?
With this PR, we can use the unittest mocking library to mock out any external SQL calls. Since we already know that the SQL calls will work, we can focus purely on whether the astro library generates the correct SQL and deletes the correct tables. This has caused a 50% reduction in run-time (16 minutes to 8 minutes).

In the interest of not making the perfect the enemy of the good, I have opted not to mock out the mapped tasks tests, as those seem more complex and I believe there are lower hanging fruit we can address to reduce CI runtime.

## Does this introduce a breaking change?
No.

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by dimberman** [comment]
@tatiana @kaxil this reduces the runtime for CI[6] from 16 minutes to 8 minutes. I have a question on how we should handle databridcks since databricks uses a different `run_sql` function so I'm not sure if there's an easy way to handle via mock or if we need a separate function.

File: python-sdk/tests_integration/sql/operators/test_cleanup.py [hunk 1]
```diff
@@ -28,164 +31,123 @@
...
-    SUPPORTED_DATABASES_OBJECTS_WITH_FILE,
-    indirect=True,
-    ids=SUPPORTED_DATABASES,
+    "temp_table",
+    [
+        Table(conn_id="sqlite_conn"),
+        Table(conn_id="snowflake_conn"),
+        Table(conn_id="bigquery"),
+    ],
+    ids=["sqlite", "snowflake", "bigquery"],
```
Comment (kaxil) [inline python-sdk/tests_integration/sql/operators/test_cleanup.py:44]: Should we add Redshift, postgres and delta too?

**Comment by kaxil** [comment]
> @tatiana @kaxil this reduces the runtime for CI[6] from 16 minutes to 8 minutes. I have a question on how we should handle databridcks since databricks uses a different `run_sql` function so I'm not sure if there's an easy way to handle via mock or if we need a separate function.

You could do something like the following:

```python
@pytest.mark.parametrize(
    "temp_table",
    [
        Table(conn_id="sqlite_conn"),
        Table(conn_id="snowflake_conn"),
        Table(conn_id="bigquery"),
        Table(conn_id="databricks_conn"),
    ],
    ids=["sqlite", "snowflake", "bigquery", "databricks"],
)
def test_cleanup_one_table(temp_table):
    module = create_database(temp_table.conn_id)
    with mock.patch(f"{module.__class__.__module__}.{module.__class__.__name__}.run_sql") as mock_run_sql:
        a = aql.cleanup([temp_table])
        a.execute({})
    mock_run_sql.assert_called_once_with(
        drop_table_statement.format("table_name", table_name=temp_table.name)
    )
```

File: python-sdk/tests_integration/sql/operators/test_cleanup.py [hunk 2]
```diff
@@ -28,164 +31,140 @@
...
-    SUPPORTED_DATABASES_OBJECTS_WITH_FILE,
-    indirect=True,
-    ids=SUPPORTED_DATABASES,
+    "temp_table",
+    [
+        Table(conn_id="sqlite_conn"),
+        Table(conn_id="snowflake_conn"),
+        Table(conn_id="bigquery"),
+        Table(conn_id="databricks_conn"),
+        Table(conn_id="redshift_conn"),
```
Comment (pankajastro) [inline python-sdk/tests_integration/sql/operators/test_cleanup.py:44]: look like Postgres is missing here. wondering why Postgres is missing in other tests too, is there any reason for this?

File: python-sdk/tests_integration/sql/operators/test_cleanup.py (hunk 1 above)
Comment (dimberman) [reply python-sdk/tests_integration/sql/operators/test_cleanup.py]: added all three

File: python-sdk/tests_integration/sql/operators/test_cleanup.py (hunk 2 above)
Comment (dimberman) [reply python-sdk/tests_integration/sql/operators/test_cleanup.py:44]: added postgres

**Review by kaxil — APPROVED** [review]

**Pull request merged**